src/librustc/ty/context.rs

```diff
@@ -882,6 +882,10 @@ pub struct GlobalCtxt<'tcx> {
     /// Maps Expr NodeId's to `true` iff `&expr` can have 'static lifetime.
     pub rvalue_promotable_to_static: RefCell<NodeMap<bool>>,
 
+    pub display_used_late_bound_region_names: RefCell<Option<FxHashSet<Name>>>,
```

Comment:
This is likely to cause problems with incremental compilation. I think the amount of  globals ppaux is depending on is getting far out of hand, and we should stop using `Display` & `Debug` and start using our own trait.